app/controllers/media/objects_controller.rb

```diff
@@ -47,7 +53,7 @@ def media_object_params
 
   def media_filename
     extension = File.extname(@media_object.item_file_name)
-    "elmo-#{@response.id}-#{@answer.id}#{extension}"
+    "elmo-#{@response.try(:id)}-#{@answer.try(:id)}#{extension}"
```

Comment:
This seems like an odd way to fail. This will generate an invalid filename if response or answer are nil, right? Shouldn't this be failing earlier on in some cleaner way if so?